### DISCUSSION TITLE
Custom JS & CSS

### DISCUSSION THREAD

**Discussion by vukcevicnikola** [gregsullivan/_tw#44] [General]
Hi Greg! First at all, thanks for this great job for WordPress and Tailwind community!

I have questions, where we can write our custom JS and CSS code? 
Thanks in advance, 
Nikola.

**Comment by gregsullivan** [comment]
Hi Nikola—you're very welcome, thanks for posting your question!

Custom JS should be placed in `./javascript/script.js`. (It will be processed with esbuild.) Custom CSS should be placed in `./tailwind/custom/`. If you're adding, e.g., the stylesheet for a JavaScript library, you can drop it into the `./tailwind/custom/components/` folder, and it will automatically be recognized and added. Otherwise, you should edit the existing CSS files used by Tailwind to generate the complete `./theme/style.css` file.

Hope that helps!

**Reply by vukcevicnikola** [reply]
Greg, that helped!  
Thanks for you quick and detailed answer.

**Reply by gregsullivan** [reply]
You're very welcome—happy to hear that helped!

**Comment by ramllo** [comment]
Hi @gregsullivan! Great Project! Following up on this matter, I've added some custom styles to the style.css sheet, but when I try to @apply some classes to different components, no changes are made. Is there anything I might be missing?

**Reply by gregsullivan** [reply]
Thanks for your question! Are you editing the `style.css` file directly? You'll need to make your changes in the `./tailwind/custom/` folder (if you aren't already). If that's not it, please let me know where you're adding the `@apply` directives, and whether new classes added as attributes to a PHP file are being added as expected.
